src/main/java/ui/TestController.java

```diff
@@ -79,4 +85,50 @@ public static boolean isCloseOnQuit() {
         return hasUI() && commandLineArgs.getOrDefault("closeonquit", "false").equalsIgnoreCase("true");
     }
 
+    /**
+     * Creates a Preferences that store data in test config file if run with test mode
+     * or in a default config file specified in the Preferences class
+     * @return
+     */
+    public static Preferences createApplicationPreference() {
+        if (isTestMode()) {
+            return createTestPreferences();
+        }
+
+        return new Preferences(Preferences.GLOBAL_CONFIG_FILE);
+    }
+
+    /**
+     * Creates a Preferences that stores data in the test directory
+     * @return
+     */
+    public static Preferences createTestPreferences() {
+        return new Preferences(Preferences.TEST_CONFIG_FILE);
+    }
+
+    /**
+     * Creates a RepoIO for the application that uses different components
+     * depending on various test options: --test, --testjson etc.
+     * @return
+     */
+    public static RepoIO createApplicationRepoIO() {
+        if (isTestMode()) {
+            return new RepoIO(Optional.of(new DummySource()),
+                              isTestJSONEnabled() ? Optional.of(new JSONStoreStub()) : Optional.empty(),
```

Comment:
I really think this should be reversed. By default (i.e. `false`) in test mode, we would like not to use real JSON (since we would like to test other units aside from JSON), Hence, we should use the stub on `false`. When we would like to incorporate JSON part in testing or would like to test the JSON component, then we will set the `--testjson=true` so that we will use real JSON component (although in test folder).